WordPress/gutenberg
# Site Editor - add query args for current context.

## Description

Initial stab at #22238.  This PR allows us to load the site editor to a specific context via query args `postType` and `postId`

Based on @vindl's comment [here](https://github.com/WordPress/gutenberg/pull/27124#discussion_r531797052), this PR has been changed from the original implementation.

The idea is that `postType` can be one of: "page", "post", "wp_template", "wp_template_part". Then the ID would be whichever post we mean to display. This covers several scenarios:

1. If we wish to link directly to a "page", we use `postType=page&postId=2`. This is enough information for the site editor to resolve the correct template to display. Same for "post"
2. If we wish to link directly to a template with no content, we can use `postType=wp_template&postId=2`. This will load the specified template without any page context set. Same for "wp_template_part".

Limitations:
1. Does not support other context types yet, such as "category" or anything related to queries.
2. If we need to resolve templates or template parts by slugs if there is no post ID, this will not work.

I think this is simple enough for a round one without having to cover the edge cases by serializing the JSON state into the URL. What do you think?

## How has this been tested?



* Load the site editor as you would normally.
* select another entity in the nav panel, verify query args are added to the url.
* reload the browser, verify the editor loads to entity corresponding to the query args.

Verify this works for various contexts including: categories, pages, posts, templates, and template parts.
Verify creating a new template updates the url as well.

## Screenshots 

## Types of changes





## Checklist:
